x-pack/plugins/lens/public/indexpattern_datasource/time_shift_utils.tsx

```diff
@@ -143,6 +143,41 @@ export function getLayerTimeShiftChecks({
   };
 }
 
+export function getPrecisionErrorWarningMessages(
+  state: IndexPatternPrivateState,
+  { activeData }: FramePublicAPI
+) {
+  const warningMessages: React.ReactNode[] = [];
+
+  if (state && activeData) {
+    const hasPrecisionError = Object.values(activeData).some((datatable) =>
+      datatable.columns.some((col) => col.meta.hasPrecisionError)
```

Comment:
“hasPrecisionError” should be part of the sourceParams of meta, not directly a part of meta. I discussed with @ppisljar and we decided for now the precision error information should be a concept of esaggs only. There should be a getter function exposed by the data plugin which checks for the source of a column and then looks up the flag from the sourceParams object